pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java

```diff
@@ -448,8 +448,13 @@ private static File extractSegmentFromFormToLocalTempFile(FormDataMultiPart form
           "Invalid multi-part for segment: %s", segmentName);
       FormDataBodyPart bodyPart = map.values().iterator().next().get(0);
 
-      File localTempFile = new File(ControllerFilePathProvider.getInstance().getFileUploadTempDir(),
-          getTempSegmentFileName(segmentName));
+      File fileUploadTempDir = ControllerFilePathProvider.getInstance().getFileUploadTempDir();
+      File localTempFile = new File(fileUploadTempDir, getTempSegmentFileName(segmentName));
+
+      if (!localTempFile.getCanonicalPath().startsWith(fileUploadTempDir.getPath())) {
+        throw new IllegalArgumentException("Invalid segment name: " + segmentName);
+      }
```

Comment:
create a file path util (put in pinot-common/src/main/java/org/apache/pinot/common/utils/FileUtils.java)
```suggestion
public static File concatAndSanitizedFile(File folderDir, File filename) {
      File filePath = new File(folderDir, filename);

      if (!filePath.getCanonicalPath().startsWith(folderDir.getPath())) {
        throw new IllegalArgumentException("Invalid file name: " + filename);
      }
    }
```
  and then the util can be used across all the following

Reply:
Makes sense. Addressed